frontend/src/constants.js

```diff
@@ -5,5 +5,11 @@ export const devicePropType = PropTypes.shape({
     name: PropTypes.string.isRequired,
     address: PropTypes.string.isRequired,
     port: PropTypes.number.isRequired,
-    state: PropTypes.oneOf(['on', 'off'])
-});
\ No newline at end of file
+    state: PropTypes.oneOf(['On', 'Off'])
+});
+
+export const groupPropType = PropTypes.shape({
```

Comment:
`trivial`
This is actually not bad, that you're avoiding code duplication by bringing this out to the constants.
However, here is a tricky part... Ok, you've defined the `GropType` which has the same meaning against all the code base of the frontend, but the gotcha is, that components in your application might have very different interfaces and it the thinner they are - the better.

Like here is an example of a simple component, which brings the fancy styling for a switch toggle which displays the status of the group:
```javascript
import { groupPropType } from './constants';

const SwitchToggle = ({ status }) => <div style={/* fancy styling */} />
SwitchToggle.propTypes = groupPropType;
```

Regardless, this is a sort of contrived example, but in a larger app you might have faced with an issues like - your `constants.js` might overgrow, or would be difficult to keep he propTypes definitions separately from the components up to date.

To summarize, this `propTypes` stuff is a sort of a meta-information, moreover, it is cut off from the prod bundle, so there is no propblem to not introduce `contanstats.js` for this purpose, but it's up to you. In your project it makes sense to me:)